libafl/src/state/mod.rs

```diff
@@ -514,6 +527,61 @@ impl<I, C, R, SC> HasStartTime for StdState<I, C, R, SC> {
     }
 }
 
+impl<I, C, R, SC> HasCurrentCorpusIdx for StdState<I, C, R, SC> {
+    fn set_corpus_idx(&mut self, idx: CorpusId) -> Result<(), Error> {
+        self.corpus_idx = Some(idx);
+        Ok(())
+    }
+
+    fn clear_corpus_idx(&mut self) -> Result<(), Error> {
+        self.corpus_idx = None;
+        Ok(())
+    }
+
+    fn current_corpus_idx(&self) -> Result<Option<CorpusId>, Error> {
+        Ok(self.corpus_idx)
+    }
+}
+
+impl<I, C, R, SC> HasStageStatus for StdState<I, C, R, SC> {
+    fn set_stage(&mut self, idx: usize) -> Result<(), Error> {
+        // ensure we are in the right frame
+        if self.stage_depth != self.stage_idx_stack.len() {
+            return Err(Error::illegal_state(
+                "stage not resumed before setting stage",
+            ));
+        }
+        self.stage_idx_stack.push(idx);
+        Ok(())
+    }
+
+    fn clear_stage(&mut self) -> Result<(), Error> {
+        self.stage_idx_stack.truncate(self.stage_depth);
```

Comment:
why truncate here?
shouldn't this function set everything in stage_idx_stack to 0?

Reply:
I saw hot it is used. it's more like `pop_last_stage(&mut self)`
can we rename this function?

Reply:
Sure, I can replace this, but I'll need to add a check for the correct stage depth.

Reply:
Ah, regarding the name, no, it is not suitable to make it `pop_last_stage` because not all implementations will support nested stages.

Reply:
how about clear_last_stage?
this sounds like it wipes off the entire things on the stack

Reply:
I think this is implementation details. I'll replace it with pop and a sanity check for depth.